music_service_async_interface/__init__.py

```diff
@@ -197,6 +197,22 @@ async def parse_urls(self, long_string: str) -> AsyncGenerator["Object", None]:
                 except InvalidURL:
                     pass
 
+    @abstractmethod
+    async def search(
+        self, query, types: Optional[Union[Type["Searchable"], List[Type["Searchable"]]]] = None
+    ) -> AsyncGenerator["Searchable", None]:
+        """Searches music service for specific type(s) of :class:`Searchable` objects using `query`
```

Comment:
either `types of objects` or `type of object`
s/query/query string/

if it refers to the Searchable class, then phrase it like: `Searches music service for specific Searchable types using....`